podcasts/PlaybackManager.swift

```diff
@@ -401,18 +401,44 @@ class PlaybackManager: ServerPlaybackDelegate {
         seekingTo != PlaybackManager.notSeeking
     }
 
-    func seekTo(time: TimeInterval, startPlaybackAfterSeek: Bool = false) {
-        seekTo(time: time, syncChanges: SyncManager.isUserLoggedIn(), startPlaybackAfterSeek: startPlaybackAfterSeek)
+    func seekTo(time: TimeInterval, startPlaybackAfterSeek: Bool = false, seekHint: SeekHint? = nil) {
+        seekTo(time: time, syncChanges: SyncManager.isUserLoggedIn(), startPlaybackAfterSeek: startPlaybackAfterSeek, seekHint: seekHint)
     }
 
     func seekToFromSync(time: TimeInterval, syncChanges: Bool, startPlaybackAfterSeek: Bool) {
         analyticsPlaybackHelper.currentSource = .sync
         seekTo(time: time, syncChanges: syncChanges, startPlaybackAfterSeek: startPlaybackAfterSeek)
     }
 
-    func seekTo(time: TimeInterval, syncChanges: Bool, startPlaybackAfterSeek: Bool = false) {
+    private var previousSeekTime: TimeInterval?
+
+    enum SeekHint {
+        case back
+        case forward
+    }
+
+    private let debouncer = Debounce(delay: 1.second)
+
+    func seekTo(time: TimeInterval, syncChanges: Bool, startPlaybackAfterSeek: Bool = false, seekHint: SeekHint? = nil) {
         guard let playingEpisode = currentEpisode() else { return } // nothing to actually seek
 
+        // When using EffectsPlayer we have an issue in which rapidly tapping
+        // skip back results (sometimes) in skipping forward
+        // Here we handle that to avoid this issue
+        // See https://github.com/Automattic/pocket-casts-ios/issues/1950
+        if seekHint == .back {
+            if let previousSeekTime, time > previousSeekTime {
```

Comment:
Why not use the currentTime value that we have bellow? And avoid the use of this variable and the deboucer?

Reply:
Because it will never work.

`currentTime` is the wrong value being given. If we compare it to it, `time > currentTime` will never succeed. You can try adding and testing it, the bug will be there.

Not sure if the usage of debouncer is clear, if not I can add a comment clarifying it. But we have the debouncer to fix issues with skip forward or other seeks after rapidly tapping skipping back happens. If we don't have it, we might get stuck into a state where seek will never work.